lib/servers/WebsocketServer.js

```diff
@@ -3,22 +3,38 @@
 /* eslint-disable
   class-methods-use-this
 */
+const url = require('url');
 const ws = require('ws');
 const BaseServer = require('./BaseServer');
 
 module.exports = class WebsocketServer extends BaseServer {
   constructor(server) {
     super(server);
     this.wsServer = new ws.Server({
-      server: this.server.listeningApp,
+      noServer: true,
       path: this.server.sockPath,
     });
 
+    this.server.listeningApp.on('upgrade', (req, sock, head) => {
+      if (!this.shouldUpgrade(req)) {
+        return;
+      }
+
+      this.wsServer.handleUpgrade(req, sock, head, (connection) => {
+        this.wsServer.emit('connection', connection, req);
+      });
+    });
+
     this.wsServer.on('error', (err) => {
       this.server.log.error(err.message);
     });
   }
 
+  shouldUpgrade(req) {
+    const pathname = url.parse(req.url).pathname;
```

Comment:
memo: we'll use URLSearchParams after major release.

Reply:
I fixed the code to use WebSocket.Server.shouldHandle()
https://github.com/websockets/ws/blob/master/doc/ws.md#servershouldhandlerequest
Should be more futureproof?